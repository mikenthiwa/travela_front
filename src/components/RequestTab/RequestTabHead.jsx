import React, { Component } from 'react';
import _ from 'lodash';
import { PropTypes } from 'prop-types';
import './RequestTabHead.scss';
import { Link } from 'react-router-dom';
import mark from '../../images/icons/new-request-icons/mark.svg';
import backButton from '../../images/back-icon.svg';


class RequestTabHead extends Component {
  constructor(props){
    super(props);
  }

  renderTab(obj, currentTab, Icon){
    const { id, name, status } = obj;
    const range = _.range(1,currentTab);
    const current = id === currentTab ? 'current_request': '';
    const completed = range.indexOf(id) !== -1? 'complete' : 'incomplete'; 
    const iconColor = range.indexOf(id) !== -1? '#10A36D' : '#3359DB'; 
    return(
      <div key={id} className={`request__tab-card mdl-cell mdl-cell--3-col mdl-cell--4-col-tablet mdl-cell--12-col-phone ${current} ${completed}`}>
        <div className="mark">
          <img src={mark} alt="" />
        </div>
        <div className="tab-logo">
          <Icon color={iconColor} />
        </div>
        <div className="tab-title">
          {name}
        </div>
        <div className="foot-text">
          {status}
        </div>
      </div>
    );
  }

  renderTitle(){
    const { editing } = this.props;
    return (
      <div className="new-request_title">
        {
          editing ? 
            (
              <div>
                <h1 className="page-header__edit-request">
               
                  <Link to="/requests"><img src={backButton} className="header__link" alt="back icon" /></Link>
                
                  <span className="edit__request-title-text">
                    {'EDIT A TRAVEL REQUEST'}
                  </span>
                </h1>
              </div>
            ):
            (
              <div>
                <span className="new__request-title-text">
                  {'CREATE A NEW TRAVEL REQUEST'}
                </span>
    
              </div>
            )
        }
      </div>
    );
  }

  render(){
    const {steps, currentTab } = this.props;
    return (
      <div>
        {this.renderTitle()}
        <div className="request__tab mdl-grid">
          {steps && steps.map(step => this.renderTab(step, currentTab, step.icon))}
        </div>
      </div>
    );  
  }

}
RequestTabHead.defaultProps = {
  editing: false,
};

RequestTabHead.propTypes = {
  currentTab: PropTypes.number.isRequired,
  steps: PropTypes.array.isRequired,
  editing: PropTypes.bool,
};


export default RequestTabHead;
