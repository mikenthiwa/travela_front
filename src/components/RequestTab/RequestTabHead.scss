.request__tab {
  margin-top: 57px;
  padding: 0;
  list-style: none;
  display: flex;
  align-items: center;
}

.request__tab-card {
  box-sizing: border-box;
  min-height: 150px;
  border-radius: 16px;
  margin-left: 0;
  margin-right: 8px;
  background-color: #ffffff;
  float: left;
  position: relative;
  padding-bottom: 23px;
  &:nth-child(4n) {
    margin-right: 0px;
  }
  .tab-logo {
    width: 45px;
    height: 45px;
    border-radius: 50%;
    margin-top: 20px;
    margin-left: auto;
    margin-right: auto;
    background: #f5f5f5;
    svg {
      left: 50%;
      width: 29px;
      transform: translate(-50%, -50%);
      float: left;
      top: 50%;
      position: relative;
    }
  }
  .tab-title {
    width: 100%;
    text-align: center;
    margin-top: 7px;
    height: 21px;
    color: #4f4f4f;
    font-family: "DIN Pro";
    font-size: 16px;

    line-height: 21px;
  }
  .foot-text {
    margin-top: 5px;
    height: 15px;
    width: 100%;
    color: #4f4f4f;
    font-family: "DIN Pro";
    font-size: 12px;
    line-height: 15px;
    text-align: center;
  }
}

.current_request {
  border: 1px solid #3359db;
  opacity: 1 !important;
  .foot-text {
    display: block !important;
  }
}

.complete {
  opacity: 1 !important;
  margin: 0;
  .tab-logo {
    background-color: rgba(16, 163, 109, 0.15);
  }
  .mark {
    display: block;
  }
}

.incomplete {
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.1);
  .foot-text {
    display: none;
  }
}

.mark {
  display: none;
  position: absolute;
  right: 10px;
  top: 10px;
  height: 24px;
  width: 24px;
  background-color: #10a36d;
  border-radius: 50%;
  text-align: center;
}

.new-request_title {
  height: 23px;
  width: 277px;
  color: #4f4f4f;
  font-family: "DIN Pro";
  font-size: 18px;

  line-height: 23px;
  margin-top: 60px;
  margin-bottom: 25px;
}
.hide_element {
  z-index: -1;
}
.page-header__edit-request {
  display: inline;
  align-items: center;
  a {
    margin-right: 5px;
  }
}

@media (max-width: 479px) {
  .request__tab-card {
    width: -webkit-stretch !important;
    margin: 8px;
  }

  .complete {
    margin-bottom: 10px;
  }
}
