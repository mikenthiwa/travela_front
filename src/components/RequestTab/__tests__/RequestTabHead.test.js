import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import RequestTabHead from '../RequestTabHead';

import tabIcons from '../../../images/icons/new-request-icons';

const props = {
  steps:[
    { id:1, name:'Personal Information', status:'', icon: tabIcons.personal },
    { id:2, name:'Trip Details', status:'', icon: tabIcons.tripDetails },
    { id:3, name:'Travel Stipends', status:'', icon: tabIcons.stipend },
    { id:4, name:'Travel Checklist', status:'', icon: tabIcons.checkList }   
  ],
  currentTab: 3,
  editing: false,

};

let wrapper;
describe('<RequestTabHead />', () => {
  beforeEach(() => {
    wrapper=  shallow (<RequestTabHead {...props} />);
  });
  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('render all the step', () =>{
    expect(wrapper.find('.request__tab-card').length).toEqual(4);
  });
  it('calls renderTab', () => {
    const renderTabSpy = jest.spyOn(wrapper.instance(), 'renderTab');
    wrapper.instance().renderTab(props.steps[0], 1, props.steps[1].icon);
    expect(renderTabSpy).toHaveBeenCalled();
  });
  it('should render the correct title for creating request', () => {
    const wrapper = mount(
      <MemoryRouter>
        <RequestTabHead {...props} />
      </MemoryRouter>);

    let title = wrapper.find('.new__request-title-text').text();
    

    expect(title).toEqual('CREATE A NEW TRAVEL REQUEST');
  });
  it('should render the correct title for edit', () => {
    const wrapper = mount(
      <MemoryRouter>
        <RequestTabHead {...props} editing />
      </MemoryRouter>);
    let title = wrapper.find('.edit__request-title-text').text();
    

    expect(title).toEqual('EDIT A TRAVEL REQUEST');

  });
});
