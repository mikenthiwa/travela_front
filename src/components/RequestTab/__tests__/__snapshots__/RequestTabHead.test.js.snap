// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestTabHead /> should render properly 1`] = `
<div>
  <div
    className="new-request_title"
  >
    <div>
      <span
        className="new__request-title-text"
      >
        CREATE A NEW TRAVEL REQUEST
      </span>
    </div>
  </div>
  <div
    className="request__tab mdl-grid"
  >
    <div
      className="request__tab-card mdl-cell mdl-cell--3-col mdl-cell--4-col-tablet mdl-cell--12-col-phone  complete"
      key="1"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <personal
          color="#10A36D"
        />
      </div>
      <div
        className="tab-title"
      >
        Personal Information
      </div>
      <div
        className="foot-text"
      />
    </div>
    <div
      className="request__tab-card mdl-cell mdl-cell--3-col mdl-cell--4-col-tablet mdl-cell--12-col-phone  complete"
      key="2"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <tripDetails
          color="#10A36D"
        />
      </div>
      <div
        className="tab-title"
      >
        Trip Details
      </div>
      <div
        className="foot-text"
      />
    </div>
    <div
      className="request__tab-card mdl-cell mdl-cell--3-col mdl-cell--4-col-tablet mdl-cell--12-col-phone current_request incomplete"
      key="3"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <stipend
          color="#3359DB"
        />
      </div>
      <div
        className="tab-title"
      >
        Travel Stipends
      </div>
      <div
        className="foot-text"
      />
    </div>
    <div
      className="request__tab-card mdl-cell mdl-cell--3-col mdl-cell--4-col-tablet mdl-cell--12-col-phone  incomplete"
      key="4"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <checkList
          color="#3359DB"
        />
      </div>
      <div
        className="tab-title"
      >
        Travel Checklist
      </div>
      <div
        className="foot-text"
      />
    </div>
  </div>
</div>
`;
